labgrid/driver/usbloader.py

```diff
@@ -79,6 +80,66 @@ def load(self, filename=None):
         mf.sync_to_resource()
 
         subprocess.check_call(
-            self.loader.command_prefix +
-            [self.tool, "-p", str(self.loader.path), "-c", mf.get_remote_path()]
+            self.loader.command_prefix
+            + [self.tool, "-p", str(self.loader.path), "-c", mf.get_remote_path()]
         )
+
+
+@target_factory.reg_driver
+@attr.s(cmp=False)
+class RKUSBDriver(Driver, BootstrapProtocol):
+    bindings = {
+        "loader": {RKUSBLoader, NetworkRKUSBLoader},
+    }
+
+    image = attr.ib(default=None)
+    usb_loader = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        # FIXME make sure we always have an environment or config
+        if self.target.env:
+            self.tool = self.target.env.config.get_tool('rk-usb-loader') or 'rk-usb-loader'
+        else:
+            self.tool = 'rk-usb-loader'
+
+    def on_activate(self):
+        pass
+
+    def on_deactivate(self):
+        pass
+
+    @Driver.check_active
+    @step(args=['filename'])
+    def load(self, filename=None):
+        if self.target.env:
+            usb_loader = self.target.env.config.get_image_path(self.usb_loader)
+            mf = ManagedFile(usb_loader, self.loader)
+            mf.sync_to_resource()
+
+        deadline = time.monotonic() + 3.0
+        while time.monotonic() < deadline:
+            try:
+                subprocess.check_call(
+                    self.loader.command_prefix +
+                    [self.tool, 'db', mf.get_remote_path()]
+                )
+                break
+            except subprocess.CalledProcessError:
+                continue
```

Comment:
If this times out, we will just continue with the next stage, without raising the exception.
With Timeout from labgrid.util, you could do:
```python
    timeout = Timeout(3.0)
    while True:
        try:
            subprocess.check_call(
                self.loader.command_prefix +
                [self.tool, 'db', mf.get_remote_path()]
            )
            break
        except subprocess.CalledProcessError:
            if timeout.expired:
                raise
```